src/client/options/mod.rs

```diff
@@ -673,41 +680,177 @@ impl Serialize for ClientOptions {
     }
 }
 
+/// Contains the options that can be set via a MongoDB connection string.
+///
+/// The format of a MongoDB connection string is described [here](https://docs.mongodb.com/manual/reference/connection-string/#connection-string-formats).
 #[derive(Debug, Default, PartialEq)]
-struct ClientOptionsParser {
-    pub hosts: Vec<ServerAddress>,
-    pub srv: bool,
+#[non_exhaustive]
+pub struct ConnectionString {
+    /// The initial list of seeds that the Client should connect to.
+    ///
+    /// Note that by default, the driver will autodiscover other nodes in the cluster. To connect
+    /// directly to a single server (rather than autodiscovering the rest of the cluster), set the
+    /// `direct_connection` field to `true`.
+    pub hosts: HostInfo,
+
+    /// The application name that the Client will send to the server as part of the handshake. This
+    /// can be used in combination with the server logs to determine which Client is connected to a
+    /// server.
     pub app_name: Option<String>,
+
+    /// The TLS configuration for the Client to use in its connections with the server.
+    ///
+    /// By default, TLS is disabled.
     pub tls: Option<Tls>,
-    pub heartbeat_freq: Option<Duration>,
+
+    /// The amount of time each monitoring thread should wait between performing server checks.
+    ///
+    /// The default value is 10 seconds.
+    pub heartbeat_frequency: Option<Duration>,
+
+    /// When running a read operation with a ReadPreference that allows selecting secondaries,
+    /// `local_threshold` is used to determine how much longer the average round trip time between
+    /// the driver and server is allowed compared to the least round trip time of all the suitable
+    /// servers. For example, if the average round trip times of the suitable servers are 5 ms, 10
+    /// ms, and 15 ms, and the local threshold is 8 ms, then the first two servers are within the
+    /// latency window and could be chosen for the operation, but the last one is not.
+    ///
+    /// A value of zero indicates that there is no latency window, so only the server with the
+    /// lowest average round trip time is eligible.
+    ///
+    /// The default value is 15 ms.
     pub local_threshold: Option<Duration>,
+
+    /// Specifies the default read concern for operations performed on the Client. See the
+    /// ReadConcern type documentation for more details.
     pub read_concern: Option<ReadConcern>,
-    pub selection_criteria: Option<SelectionCriteria>,
+
+    /// The name of the replica set that the Client should connect to.
     pub repl_set_name: Option<String>,
```

Comment:
Along the lines of the other option, should we name this `replica_set` to match the actual URI option? We could also just document that it corresponds to the `replicaSet` option.

Reply:
Renamed.